native/cocos/renderer/pipeline/ReflectionProbeBatchedQueue.cpp

```diff
@@ -77,21 +78,30 @@ void ReflectionProbeBatchedQueue::gatherRenderObjects(const scene::Camera *camer
     const scene::Skybox *skyBox = sceneData->getSkybox();
     const scene::RenderScene *const scene = camera->getScene();
 
-    if (static_cast<uint32_t>(camera->getClearFlag()) & skyboxFlag) {
+    if (static_cast<uint32_t>(probe->getCamera()->getClearFlag()) & skyboxFlag) {
         if (skyBox != nullptr && skyBox->isEnabled() && skyBox->getModel()) {
             add(skyBox->getModel());
         }
     }
     for (const auto &model : scene->getModels()) {
         const auto *node = model->getNode();
-        if (!node) continue;
-        if (((node->getLayer() & REFLECTION_PROBE_DEFAULT_MASK) == node->getLayer())
-            || (REFLECTION_PROBE_DEFAULT_MASK & static_cast<uint32_t>(model->getVisFlags()))) {      
-            if (model->getWorldBounds()) {
-                if (model->getWorldBounds()->aabbFrustum(probe->getCamera()->getFrustum())) {
+        if (!node || !model->isEnabled() || !model->getWorldBounds() || !model->getBakeToReflectionProbe()) continue;
+        uint32_t visibility = probe->getCamera()->getVisibility();
+        if (probe->getProbeType() == scene::ReflectionProbe::ProbeType::CUBE) {
+            if (((visibility & node->getLayer()) == node->getLayer()) ||
+                (visibility & static_cast<uint32_t>(model->getVisFlags()))) {
+                if (aabbWithAABB(*model->getWorldBounds(), *probe->getBoundingBox())) {
                     add(model);
                 }
             }
+        } else {
+            if (((node->getLayer() & REFLECTION_PROBE_DEFAULT_MASK) == node->getLayer()) || (REFLECTION_PROBE_DEFAULT_MASK & static_cast<uint32_t>(model->getVisFlags()))) {
+                if (model->getWorldBounds()) {
+                    if (model->getWorldBounds()->aabbFrustum(probe->getCamera()->getFrustum())) {
```

Comment:
It is better to define a variable to hold `model->getWorldBounds()` as it is used many times in a loop.